rllib/core/optim/tests/test_rl_optimizer.py

```diff
@@ -0,0 +1,386 @@
+            obs = next_obs
+        batch = SampleBatch(
+            {
+                "obs": _obs,
+                "next_obs": _next_obs,
+                "actions": _actions,
+                "rewards": _rewards,
+                "dones": _dones,
+            }
+        )
+        _returns.append(_return)
+        _rollouts.append(batch)
+    return np.mean(_returns), _returns, _rollouts
+
+
+class BCTorchModule(TorchRLModule):
+    def __init__(self, config: FCConfig) -> None:
+        super().__init__(config)
+        self.policy = FCNet(config)
+
+    def input_specs(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+                "actions": TorchTensorSpec("b"),
+            }
+        )
+
+    def output_specs(self) -> ModelSpec:
+        return ModelSpec({"action_dist": Distribution})
+
+    def input_specs_exploration(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_inference(self) -> ModelSpec:
+        obs_dim = self.config.input_dim
+        return ModelSpec(
+            {
+                "obs": TorchTensorSpec("b, do", do=obs_dim),
+            }
+        )
+
+    def input_specs_train(self) -> ModelSpec:
+        return self.input_specs()
+
+    def output_specs_exploration(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_inference(self) -> ModelSpec:
+        return self.output_specs()
+
+    def output_specs_train(self) -> ModelSpec:
+        return self.output_specs()
+
+    def _forward_inference(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        with torch.no_grad():
+            action_logits = self.policy(obs)
+        action_logits_inference = torch.argmax(action_logits, dim=-1)
+        action_dist = TorchDeterministic(action_logits_inference)
+        return {"action_dist": action_dist}
+
+    def _forward_exploration(self, batch: NestedDict) -> Mapping[str, Any]:
+        with torch.no_grad():
+            return self._forward_train(batch)
+
+    def _forward_train(self, batch: NestedDict) -> Mapping[str, Any]:
+        obs = batch[SampleBatch.OBS]
+        action_logits = self.policy(obs)
+        action_dist = TorchCategorical(logits=action_logits)
+        return {"action_dist": action_dist}
+
+    def is_distributed(self) -> bool:
+        return False
+
+    def make_distributed(self, dist_config: Mapping[str, Any] = None) -> None:
+        pass
+
+    def get_state(self) -> Mapping[str, Any]:
+        return {"policy": self.policy.state_dict()}
+
+    def set_state(self, state: Mapping[str, Any]) -> None:
+        self.policy.load_state_dict(state["policy"])
+
+
+class BCTorchOptimizer(RLOptimizer):
+    def __init__(self, module, config):
```

Comment:
not specifically about this PR. we may be abusing name config a bit.
just in this file, I am seeing config used for model, RLModule, and optimizer.
maybe we standardize, and say configs that are being passed around are always the full config object?
and the first line of a function can do ``config.model`` or ``config.model.optimizer``.

if we want to pass specific config obj into different places, let's rename the variable to optimzier_config, model_config, etc?